lang/cs/Org.Apache.REEF.Network/Group/Driver/Impl/CommunicationGroupDriver.cs

```diff
@@ -119,12 +121,23 @@ public ICommunicationGroupDriver AddBroadcast<T>(string operatorName, string mas
                 topology = new FlatTopology<T>(operatorName, _groupName, spec.SenderId, _driverId,
                     spec);
             }
-            else
+            else if(topologyType == TopologyTypes.Tree)
             {
                 topology = new TreeTopology<T>(operatorName, _groupName, spec.SenderId, _driverId,
                     spec,
                     _fanOut);
             }
+            else if (topologyType == TopologyTypes.WritableFlat)
```

Comment:
I mean the entire if/else portion can be moved to a method called CreateTopology() and can be shared by AddBroadcast, AddReduce, AddScatter,... if they are the same. Of course, if it is different in each method, then leave it as it is.